tests/functional/object/mcg/test_obj_owner_md.py

```diff
@@ -0,0 +1,232 @@
+        self,
+        mcg_obj,
+        other_acc_creds,
+        bucket_factory,
+        awscli_pod_session,
+        test_directory_setup,
+    ):
+        """
+        Test the ownership behavior of objects in S3 buckets.
+        Objects in a bucket should be owned by the account that created the bucket:
+
+        1. Create a bucket using the noobaa admin account credentials
+        2. Allow the other account access to the admin's bucket
+        3. Create a bucket using another account credentials
+        4. Write objects from each account to both buckets
+        5. For both buckets, check that the objects are owned by the creator of the bucket
+
+        """
+        # 1. Create a bucket using the noobaa admin account credentials
+        admin_bucket_name = bucket_factory()[0].name
+
+        # 2. Allow the other account access to the admin's bucket
+        bucket_policy = gen_bucket_policy(
+            user_list="*",
+            actions_list=["*"],
+            resources_list=[admin_bucket_name, f"{admin_bucket_name}/*"],
+        )
+        put_bucket_policy(mcg_obj, admin_bucket_name, json.dumps(bucket_policy))
+
+        # Set mcg_obj to use the other account's credentials
+        mcg_obj.s3_resource = boto3.resource(
+            "s3",
+            verify=False,
+            endpoint_url=mcg_obj.s3_endpoint,
+            aws_access_key_id=other_acc_creds["access_key_id"],
+            aws_secret_access_key=other_acc_creds["access_key"],
+        )
+        mcg_obj.s3_client = mcg_obj.s3_resource.meta.client
+
+        # 3. Create a bucket using another account credentials
+        # Since bucket_factory uses the MCG object,
+        # the following command creates a bucket using the other account's creds
+        non_admin_bucket_name = bucket_factory()[0].name
+
+        # Set the mcg_obj credentials back to the noobaa admin account's
+        mcg_obj.update_s3_creds()
+
+        # 4. Write objects from each account to both buckets
+        for bucket in (admin_bucket_name, non_admin_bucket_name):
+            write_random_test_objects_to_bucket(
+                amount=3,
+                io_pod=awscli_pod_session,
+                file_dir=test_directory_setup.origin_dir,
+                pattern="admin-obj",
+                bucket_to_write=bucket,
+                mcg_obj=mcg_obj,
+            )
+
+            # Using the non-privileged account's credentials requires
+            # the use of retry since the bucket policy may take some time to propagate
+            retry_write_random_objs = retry(CommandFailed, tries=10, delay=10)(
+                write_random_test_objects_to_bucket
+            )
+            retry_write_random_objs(
+                amount=3,
+                io_pod=awscli_pod_session,
+                file_dir=test_directory_setup.origin_dir,
+                pattern="non-admin-obj",
+                bucket_to_write=bucket,
+                s3_creds=other_acc_creds,
+            )
```

Comment:
@sagihirshfeld Can we optimize this block by writing one more for loop which will iterate 2 times for privileged user and non-privileged user

Reply:
Addressed in https://github.com/red-hat-storage/ocs-ci/pull/10175/commits/24f1111543f610c262ef8f9befdfc4ad2d7356bc